framework/lib/components/text-field/formatted-number-input.tsx

```diff
@@ -75,28 +76,41 @@ export const FormattedNumberInput = ({
   preset = 'eu',
   isPercentage = false,
   onChange = identity,
-  value,
+  value: valueProp,
   numberOfDecimals = 2,
   max = Infinity,
   min = -Infinity,
   inputLeftElement,
   inputRightElement,
+  onBlur,
   ...rest
 }: FormattedNumberInputProps) => {
+  const [ valueState, setValueState ] = useState(valueProp)
+  const isControlled = typeof valueProp !== 'undefined'
+  const value = isControlled ? valueProp : valueState
   const props = presetMap[preset]
-  const [ v, setV ] = useState(value)
 
   const validateRange = () => {
-    if (isNil(v)) return
-    const vNum = typeof v === 'string' ? parseFloat(v) : v
+    if (isNil(value)) return
+    const vNum = typeof value === 'string' ? parseFloat(value) : value
     const factor = isPercentage ? 100 : 1
     if (vNum * factor > max) {
       const newValue = roundToPrecision(max / factor, numberOfDecimals)
-      setV(newValue)
+      setValueState(newValue)
+      onChange({
+        floatValue: newValue,
+        formattedValue: numericFormatter(newValue.toString(), props),
+        value: newValue.toString(),
+      })
```

Comment:
Maybe put in separate function, getNumberFormatValueObject or similiar? This is a preference detail tho so feel free to skip